### PR TITLE
Add FastQC example; small updates to samtools/compress examples

### PR DESCRIPTION
**Pull request by mbookman**

### CODE REVIEW

**Comment by deflaux** [comment]
LGTM, nice sample, I like the demonstration for multiple input files

File: fastqc/cloud/run_fastqc.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+      # The Pipelines API will create the input directory when localizing files,
+      # but does not create the output directory.
+      'cmd': ('mkdir /mnt/data/output && '
+              'fastqc /mnt/data/input/* --outdir=/mnt/data/output/'),
+    },
+
+    # The Pipelines API currently supports full GCS paths, along with patterns (globs),
+    # but it doesn't directly support a list of files being passed as a single input
+    # parameter ("gs://bucket/foo.bam gs://bucket/bar.bam").
```
Comment (pgrosu) [inline fastqc/cloud/run_fastqc.py]: Why is this not an easy fix in the JES code-base to make it an `ArrayList<Paths>`, in this [portion of code](https://github.com/broadinstitute/cromwell/blob/6cf80782b1bb15d0ad1bbc0b8e0f4105248f0fad/src/main/scala/cromwell/engine/backend/jes/JesAttachedDisk.scala#L87-L91):

```Scala
case class JesWorkingDisk(diskType: DiskType, sizeGb: Int) extends JesAttachedDisk {
  val mountPoint = Paths.get(JesWorkingDisk.MountPoint)
  val name = JesWorkingDisk.Name
  override def toString: String = s"$name $sizeGb ${diskType.diskTypeName}"
}
```

Comment (mbookman) [inline fastqc/cloud/run_fastqc.py]: That is not the JES code-base. That is Cromwell's use of JES.
I have already filed a feature request with the Google Engineering team to support a list of paths.

Comment (pgrosu) [inline fastqc/cloud/run_fastqc.py]: That sounds reasonable, but are you not in control of what gets committed to [google-api-services-genomics](https://oss.sonatype.org/#nexus-search;quick~google-api-services-genomics)?


There is already a `List<Disk>` in `PipelineResources.java`, to which a `List<Path>` or `List<GCSPath>` would be very simple to just add:

```Java
 /**
   * Disks to attach.
   * The value may be {@code null}.
   */
  @com.google.api.client.util.Key
  private java.util.List<Disk> disks;
```

Am I simplifying things too much, or are things more complicated than that?

Thanks,
~p

Comment (pgrosu) [inline fastqc/cloud/run_fastqc.py]: @mbookman  I started issue https://github.com/googlegenomics/pipelines-api-examples/issues/13 to continue our discussion.

**Pull request merged**